esmvaltool/cmorizers/obs/cmorize_obs_era_interim.py

```diff
@@ -0,0 +1,329 @@
+    # Add height coordinate to tas variable (required by the new backend)
+    if 'height2m' in definition.dimensions:
+        utils.add_scalar_height_coord(cube, 2.)
+    if 'height10m' in definition.dimensions:
+        utils.add_scalar_height_coord(cube, 10.)
+    for axis in 'T', 'X', 'Y', 'Z':
+        coord_def = definition.coordinates.get(axis)
+        if coord_def:
+            coord = cube.coord(axis=axis)
+            if axis == 'T':
+                coord.convert_units('days since 1850-1-1 00:00:00.0')
+            if axis == 'Z':
+                coord.convert_units(coord_def.units)
+            coord.standard_name = coord_def.standard_name
+            coord.var_name = coord_def.out_name
+            coord.long_name = coord_def.long_name
+            coord.points = coord.core_points().astype('float64')
+            if len(coord.points) > 1:
+                coord.guess_bounds()
+
+
+def _fix_frequency(cube, var):
+    # Here var_name is the CMIP name
+    # era-interim is in 3hr or 6hr or 12hr freq need to convert to daily
+    # only variables with step 12 need accounting time 00 AM as time 24 PM
+    if var['mip'] in {'day', 'Eday', 'CFday'}:
+        # accounting time 00 AM as time 24 PM
+        if cube.var_name in {'tasmax', 'tasmin', 'pr',
+                             'rsds', 'hfds', 'evspsbl',
+                             'rsdt', 'rss', 'prsn'}:
+            cube.coord('time').points = [
+                cube.coord('time').units.date2num(
+                    cell.point - timedelta(seconds=1)
+                )
+                for cell in cube.coord('time').cells()
+            ]
+        if cube.var_name == 'tasmax':
+            cube = daily_statistics(cube, 'max')
+        elif cube.var_name == 'tasmin':
+            cube = daily_statistics(cube, 'min')
+        elif cube.var_name in {'pr', 'rsds', 'hfds', 'evspsbl',
+                               'rsdt', 'rss', 'prsn'}:
+            # Sum is not available in daily_statistics so call iris directly
+            if not cube.coords('day_of_year'):
+                iris.coord_categorisation.add_day_of_year(cube, 'time')
+            if not cube.coords('year'):
+                iris.coord_categorisation.add_year(cube, 'time')
+            cube = cube.aggregated_by(['day_of_year', 'year'],
+                                      iris.analysis.SUM)
+        else:
+            cube = daily_statistics(cube, 'mean')
+        # Remove daily statistics helpers
+        cube.remove_coord(cube.coord('day_of_year'))
+        cube.remove_coord(cube.coord('year'))
+        # Correct the time bound
+        cube.coord('time').points = cube.coord('time').units.date2num(
+            [
+                cell.point.replace(hour=12, minute=0, second=0,
+                                   microsecond=0)
+                for cell in cube.coord('time').cells()
+            ]
+        )
+        cube.coord('time').bounds = None
+        cube.coord('time').guess_bounds()
+    return cube
```

Comment:
Are the time coordinates for the montly values already correct? I see only code for daily values. I would at the very least expect this code to also add bounds for monthly values.

Reply:
@bouweandela thanks. These two lines are moved to line numbers 276-277. Also, some useful links are added to the documentation in lines 77-80.